airbyte-cdk/bulk/core/load/src/test/kotlin/io/airbyte/cdk/load/task/DestinationTaskLauncherUTest.kt

```diff
@@ -0,0 +1,135 @@
+
+    // Checkpoint Tasks
+    private val flushCheckpointsTaskFactory: FlushCheckpointsTaskFactory = mockk(relaxed = true)
+    private val timedFlushTask: TimedForcedCheckpointFlushTask = mockk(relaxed = true)
+    private val updateCheckpointsTask: UpdateCheckpointsTask = mockk(relaxed = true)
+
+    // Exception handling
+    private val exceptionHandler: TaskExceptionHandler<LeveledTask, WrappedTask<ScopedTask>> =
+        mockk(relaxed = true)
+
+    private fun getDefaultDestinationTaskLauncher(
+        useFileTranfer: Boolean
+    ): DefaultDestinationTaskLauncher {
+        return DefaultDestinationTaskLauncher(
+            taskScopeProvider,
+            catalog,
+            syncManager,
+            inputConsumerTask,
+            spillToDiskTaskFactory,
+            setupTaskFactory,
+            openStreamTaskFactory,
+            processRecordsTaskFactory,
+            processFileTaskFactory,
+            processBatchTaskFactory,
+            closeStreamTaskFactory,
+            teardownTaskFactory,
+            flushCheckpointsTaskFactory,
+            timedFlushTask,
+            updateCheckpointsTask,
+            exceptionHandler,
+            useFileTranfer
+        )
+    }
+
+    // private val mainThread = newSingleThreadContext("UI thread")
+
+    @Test
+    fun `test that we don't start the spill-to-disk task when file transfer is enabled`() =
+        runTest {
+            coEvery { exceptionHandler.withExceptionHandling(any()) } returns mockk(relaxed = true)
+            coEvery { taskScopeProvider.launch(any()) } returns Unit
+
+            val destinationTaskLauncher = getDefaultDestinationTaskLauncher(true)
+            // This is needed to let the run method to complete
+            destinationTaskLauncher.handleTeardownComplete()
+            destinationTaskLauncher.run()
+
+            coVerify { catalog wasNot Called }
+            coVerify { spillToDiskTaskFactory wasNot Called }
+        }
+
+    @Test
+    fun `test that we start the spill-to-disk task when file transfer is disabled`() = runTest {
+        coEvery { exceptionHandler.withExceptionHandling(any()) } returns mockk(relaxed = true)
+        coEvery { taskScopeProvider.launch(any()) } returns Unit
+
+        val stream = mockk<DestinationStream>(relaxed = true)
+        val streamDescriptor = DestinationStream.Descriptor("namespace", "name")
+        every { stream.descriptor } returns streamDescriptor
+        coEvery { catalog.streams } returns listOf(stream)
+
+        val spillToDiskTask = mockk<SpillToDiskTask>(relaxed = true)
+        coEvery { spillToDiskTaskFactory.make(any(), any()) } returns spillToDiskTask
+
+        val destinationTaskLauncher = getDefaultDestinationTaskLauncher(false)
+        // This is needed to let the run method to complete
+        destinationTaskLauncher.handleTeardownComplete()
+        destinationTaskLauncher.run()
+
+        coVerify { spillToDiskTaskFactory.make(any(), any()) }
```

Comment:
Technically this just verifies that the task is created. Technically the contract is that it passes through TaskExceptionHandler.withExceptionHandling and TaskScopeProvider.launch